pkg/expr/func_arr.go

```diff
@@ -220,3 +222,81 @@ func slice(arr interface{}, start, end int) interface{} {
 
 	return v.Slice(start, end).Interface()
 }
+
+// sortSlice sorts slice
+func sortSlice(arr interface{}, desc bool) (out interface{}, err error) {
+	if arr, err = toSlice(arr); err != nil {
+		return
+	}
+
+	// sort slice of native type values
+	var (
+		v  = reflect.ValueOf(arr)
+		vi = v.Interface()
+	)
+
+	// sort the array if error is nil;
+	// we trap the error(s) in the outer scope from the less function,
+	// and returning the last error at a time since all error will the same in almost every scenario.
+	out, sortErr := sortSliceStable(vi, func(i, j int) bool {
+		var (
+			nVal = v.Index(i)
+			mVal = v.Index(j)
+		)
+		if desc {
+			nVal = v.Index(j)
+			mVal = v.Index(i)
+		}
+
+		switch getKind(vi) {
+		case reflect.String:
+			return nVal.String() < mVal.String()
+		case reflect.Int, reflect.Int32, reflect.Int64:
+			return nVal.Int() < mVal.Int()
+		case reflect.Float32, reflect.Float64:
+			return nVal.Float() < mVal.Float()
+		default:
+			// sort slice of typed values
+			if stv, is := arr.([]TypedValue); is {
+				if casted, ok := stv[i].(Comparable); ok {
+					var c int
+					c, err = casted.Compare(stv[j])
+					if err != nil {
+						return false
+					}
+					if desc {
+						return c > 0
+					} else {
+						return c < 0
+					}
+				} else {
+					err = fmt.Errorf("cannot compare %s and %s: unknown state", stv[i].Type(), stv[j].Type())
+					return false
+				}
+			}
+			err = fmt.Errorf("cannot compare %s and %s: unknown state", getKind(nVal), getKind(mVal))
+			return false
+		}
+	})
+	if sortErr != nil {
+		return arr, err
+	}
+
+	// returns error from less function
+	if err != nil {
+		return arr, err
+	}
+
+	return
+}
+
+// sortStable sorts the slice x using the provided less
+// function, keeping equal elements in their original order.
+func sortSliceStable(arr interface{}, less func(i, j int) bool) (out interface{}, err error) {
+	if arr, err = toSlice(arr); err != nil {
+		return
+	}
+
+	sort.SliceStable(arr, less)
+	return arr, err
+}
```

Comment:
I would personally remove this function and merge the contents to the caller as there is only one call; up to you